### PR TITLE
Feature/redbox 367 show citation text

### PR DESCRIPTION
**Pull request by brunns**
## Context

Allow the user to see some detail of the citations.

## Changes proposed in this pull request

Screenshot 2024-07-01 at 13 12 38

Screenshot 2024-07-01 at 13 12 48

## Relevant links

https://technologyprogramme.atlassian.net/browse/REDBOX-367

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [x] I have run integration tests

### CODE REVIEW

File: django_app/redbox_app/redbox_core/models.py
Comment (rachaelcodes) [inline django_app/redbox_app/redbox_core/models.py:213]: Would `created_at` be a better metric to sort by? Or does the order not matter?

Comment (brunns) [reply django_app/redbox_app/redbox_core/models.py:213]: Right now, it doesn't matter - we only need a total ordering so that we can group citations by file. If one day we need some specific order, we can revisit.

**Review by rachaelcodes — COMMENTED** [review]
One problem of this approach is that the order of the source responses is lost on the citations page. 

e.g. on the chat view - which has the most relevant source file listed first: 
chat_sources

and in the citations page view - which loses this ordering
sources_with_citations

Given that this is the 'quick' initial round of showing citations, I would be happy with this discrepancy if Charlotte and Gabe are.

File: django_app/redbox_app/redbox_core/models.py
Comment (rachaelcodes) [reply django_app/redbox_app/redbox_core/models.py:213]: I've put in another comment about the grouping by file - can we preserve the order of the 'most relevant file'? Or is that for a later PR?

Comment (brunns) [reply django_app/redbox_app/redbox_core/models.py:213]: Right now we don't store the citation order at all AFAICT. We might need to revisit that.

**Review by rachaelcodes — APPROVED** [review]
A good quick pass at this, which I think will add a lot of user value. Approved pending approval from Gabe/Charlotte re the discrepancy in File order.

**Pull request merged**
